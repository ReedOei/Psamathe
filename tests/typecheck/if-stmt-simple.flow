contract Main {}
;

if 1 = 1 { pass } else { pass }

