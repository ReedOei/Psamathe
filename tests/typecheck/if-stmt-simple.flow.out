  <generatedTop>
    <common>
      <k>
        .
      </k>
      <contracts>
        Main |-> contract Main { .Decls }
      </contracts>
      <fields>
        Main |-> .Set
      </fields>
      <constructors>
        Main |-> constructor ( .VarDefs , pass )
      </constructors>
      <transactions>
        .Map
      </transactions>
      <types>
        address |-> modified ( SetItem ( consumable )
        SetItem ( nonfungible ) , address )
        bool |-> modified ( SetItem ( consumable )
        SetItem ( nonfungible ) , bool )
        nat |-> modified ( SetItem ( consumable )
        SetItem ( fungible ) , nat )
        string |-> modified ( SetItem ( consumable )
        SetItem ( nonfungible ) , string )
      </types>
      <freshCounter>
        0
      </freshCounter>
    </common>
    <typeEnv>
      .Map
    </typeEnv>
  </generatedTop>
#And
  {
    ?Delta
  #Equals
    .Map
  }
#And
  {
    ?Gamma0
  #Equals
    .Map
  }
#And
  {
    ?Gamma
  #Equals
    .Map
  }
#And
  {
    ?Tau
  #Equals
    ! nat
  }
#And
  {
    ?Xi
  #Equals
    .Map
  }
